daemon/api.go

```diff
@@ -2519,6 +2524,24 @@ func postDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 		}, nil)
 	case "can-manage-refreshes":
 		return SyncResponse(devicestate.CanManageRefreshes(st), nil)
+	case "connectivity":
+		s := snapstate.Store(st)
+		st.Unlock()
+		checkResult, err := s.ConnectivityCheck()
+		st.Lock()
+		if err != nil {
+			return InternalError("cannot run connectivity check: %v", err)
+		}
+		status := ConnectivityStatus{Connectivity: true}
+		for host, reachable := range checkResult {
+			if !reachable {
+				status.Connectivity = false
+				status.Unreachable = append(status.Unreachable, host)
+			}
+		}
+		sort.Strings(status.Unreachable)
+
+		return SyncResponse(status, nil)
```

Comment:
In case of DNS issues this could easily take N*several-seconds, should this be async?

Reply:
I think it is ok if it is sync, when I run e.g. ping I get the same kind of blocking behaviour.